kahuna/public/js/components/gr-panel/gr-panel.js

```diff
@@ -60,67 +75,127 @@ grPanel.controller('GrPanel', [
+        const archivedState$ = Rx.Observable.combineLatest(
+            archivedCount$,
+            selection.count$,
+            (archivedCount, selectedCount) => {
+                switch (archivedCount) {
+                case 0:              return 'unarchived';
+                case selectedCount:  return 'archived';
+                default:             return 'mixed';
+                }
+            }
+        );
+        inject$($scope, archivedCount$, ctrl, 'archivedCount');
+        inject$($scope, archivedState$, ctrl, 'archivedState');
+
+        const selectedLabels$ = selectedImagesList$.
+              map(imageList.getLabels).
+              map(imageList.getOccurrences);
+        inject$($scope, selectedLabels$, ctrl, 'selectedLabels');
+
+        const selectedUsageRights$ = selectedImagesList$.map(selectedImagesList => {
+            // FIXME: wrap into slightly weird shape expected by usage
+            // rights editor component
+            return selectedImagesList.map(image => {
+                return {
+                    image: image,
+                    data: imageAccessor.readUsageRights(image)
+                };
             });
-        };
+        });
+        const selectedUsageRightsArray$ = selectedUsageRights$.map(selectedUsageRights => {
+            return selectedUsageRights.toArray();
+        });
+
+        const categoriesPromise = editsApi.getUsageRightsCategories();
+        const usageRightsCategories$ = Rx.Observable.fromPromise(categoriesPromise);
+        const selectedUsageCategory$ = Rx.Observable.combineLatest(
+            selectedUsageRights$,
+            usageRightsCategories$,
+            (usageRights, categories) => {
+                const categoryCode = usageRights.reduce((m, o) => {
+                    return (m == o.data.category) ? o.data.category : 'multiple categories';
+                }, usageRights.first() && usageRights.first().data.category);
 
-        $scope.$watch(() => selection.getMetadata(), onValChange(newMetadata => {
-            ctrl.rawMetadata = newMetadata;
-            ctrl.images = Array.from(ctrl.selectedImages);
+                const usageCategory = categories.find(cat => cat.value === categoryCode);
+                return usageCategory ? usageCategory.name : categoryCode;
+            }
+        );
 
-            if (ctrl.images.length > 0) {
-                panelService.available(panelName, false);
-            } else {
+        inject$($scope, selectedUsageRightsArray$, ctrl, 'usageRights');
+        inject$($scope, selectedUsageCategory$,    ctrl, 'usageCategory');
+
+
+        // FIXME: distinct?
+        const selectedMetadata$ = selectedImagesList$.
+              map(imageList.getMetadata).
+              map(imageList.getSetOfProperties);
+        const rawMetadata$ = selectedMetadata$.map(selectedMetadata => {
+            return selectedMetadata.map((values) => {
+                switch (values.size) {
```

Comment:
I think I carried that over from the previous version of this code, and I find it reads quite well. Except I noticed a bug in my copy, so thanks :-)